elastisys/compliantkubernetes
# Auto-generate docs from JSON schema

⚠️ IMPORTANT ⚠️: This is a public repository. Make sure to not disclose:

- [x] personal data beyond what is necessary for interacting with this Pull Request;
- [x] business confidential information, such as customer names.

Quality gates:

- [x] I'm aware of the [Contributor Guide](CONTRIBUTING.md) and did my best to follow the guidelines.
- [x] I'm aware of the [Glossary](docs/glossary.md) and did my best to use those terms.

Adds auto-generate docs from the JSON schemas in Apps.

_Since it currently generates 646 files I'm just gonna let this be a thing in the deployment workflow, with optional steps to view it locally._

To test it out:

> [!important]
> You need to have `yq` version 4 installed as `yq`.

```bash
npm install
./scripts/jsonschema2md.sh
mkdocs serve
```

And then it should be available under:
- "For Platform Administrators"
  - "Configuration Reference" 
    - "Index" 
    - "Config"
    - "Secrets"

> [!tip]
> To preview for another branch in Apps set the env `GITHUB_REF_NAME` to the name of the branch.

If you set `release-vX.Y` it should do the _release flow_.
So this _should just work_ for releases but we'll have to see about that for v0.39.

Screenshot:
![image](https://github.com/elastisys/compliantkubernetes/assets/58822152/3a1c6855-7f4b-46a1-b7df-6dba6b4116f2)


Closes #852
